src/System.Security.Cryptography.Algorithms/tests/CryptoConfigTests.cs

```diff
@@ -20,15 +20,53 @@ public static void AllowOnlyFipsAlgorithms()
         }
 
         [Fact]
-        public static void AddOID_NotSupported()
+        public static void AddOID_MapNameToOID_ReturnsMapped()
         {
-            Assert.Throws<PlatformNotSupportedException>(() => CryptoConfig.AddOID(string.Empty, string.Empty));
+            CryptoConfig.AddOID("1.3.14.3.2.28", "SHAFancy");
+            Assert.Equal("1.3.14.3.2.28", CryptoConfig.MapNameToOID("SHAFancy"));
         }
 
         [Fact]
-        public static void AddAlgorithm_NotSupported()
+        public static void AddOID_EmptyString_Throws()
         {
-            Assert.Throws<PlatformNotSupportedException>(() => CryptoConfig.AddAlgorithm(typeof(CryptoConfigTests), string.Empty));
+            Assert.Throws<ArgumentException>(() => CryptoConfig.AddOID(string.Empty, string.Empty));
+        }
+
+        [Fact]
+        public static void AddOID_NullOid_Throws()
+        {
+            Assert.Throws<ArgumentNullException>(() => CryptoConfig.AddOID(null, string.Empty));
+        }
+
+        [Fact]
+        public static void AddOID_NullNames_Throws()
```

Comment:
@ianhays Do we want to throw in that case? If we pass an empty array of names there's no observable effect that we can assert on with the current implementation.